source/languages/en/riak/dev/taste-of-riak/object-modeling-csharp.md

```diff
@@ -0,0 +1,95 @@
+time they can be made unique enough for a KV store.
+
+| Bucket | Key Pattern | Example Key
+|:-------|:------------|:-----------
+| `Users` | `<user_name>` | `joeuser`
+| `Msgs` | `<username>_<datetime>` | `joeuser_2014-03-06T02:05:13`
+| `Timelines` | `<username>_<type>_<date>` | `joeuser_Sent_2014-03-06`<br /> `marketing_group_Inbox_2014-03-06` |
+
+For the `Users` bucket, we can be certain that we will want each
+username to be unique, so let's use the `username` as the key.
+
+For the `Msgs` bucket, let's use a combination of the username and the
+posting UTC datetime in an [ISO 8601](http://en.wikipedia.org/wiki/ISO_8601) format.
+This combination gives us the pattern `<username>_<datetime>`, which produces keys like
+`joeuser_2014-03-05T23:20:28`.
+
+Now for `Timelines`, we need to differentiate between `Inbox` and `Sent`
+timelines, so we can simply add that type into the key name. We will
+also want to partition each collection object into some time period,
+that way the object doesn't grow too large (see note below).
+
+For `Timelines`, let's use the pattern `<username>_<type>_<date>` for
+users, and `<groupname>_Inbox_<date>` for groups, which will look like
+`joeuser_Sent_2014-03-06` or `marketing_group_Inbox_2014-03-05`,
+respectively.
+
+<div class="note">
+<div class="title">Note</div>
+Riak performs best with objects under 1-2MB. Objects larger than that
+can hurt performance, especially many siblings are being created. We
+will cover siblings, sibling resolution, and sibling explosions in the
+next chapter.  </div>
+
+#### Keeping our story straight with repositories
+
+Now that we've figured out our object model, please refer to [this source code](https://github.com/basho/taste-of-riak/tree/master/csharp/Ch03-Msgy-Schema/Repositories) for the repositories that we'll be using.
+
+[This console application](https://github.com/basho/taste-of-riak/blob/master/csharp/Ch03-Msgy-Schema/Program.cs) exercises the code that we've written.
+
+The repository pattern and `TimelineManager` help with a few things:
+
+ - It helps us to see if an object exists before creating a new one
+ - It keeps our buckets and key names consistent
+ - It provides us with a consistent interface to work with.
+
+While this set of repositories solves many of our problems, it is very
+minimal and doesn't cover all the edge cases. For instance, what happens
+if two different people try to create a user with the same username?
+
+We can also easily "compute" key names now, but how do we quickly look
+up the last 10 messages a user sent? Many of these answers will be
+application dependent. If your application shows the last 10 messages in
+reverse order, for example, you may want to store that set of data in
+another collection object to make lookup faster. There are drawbacks to
+every solution, but we recommend seeking out the key/value-based
+solution first, as it will likely be the quickest.
+
+So to recap, in this chapter we learned:
```

Comment:
Nope!

Seriously though, if the other's don't have it, I'd prefer to keep things consistent.